### DISCUSSION TITLE
Unable to detect Python Language on repository

### DISCUSSION THREAD

**Discussion by abentgeorge** [github-linguist/linguist#5162] [Classification] [answered]
## 
I'm practicing data cleaning via python. However my [repository](https://github.com/abentgeorge/Data_Cleaning---Splitting-Names) is only detecting JupyterNotebook 

I understand why Jupyter notebook is being detected(extra cell code) but couldn't find a workaround

I tried creating the python file on Sublime text and uploading it separately but even then the language is only detecting Jupyter.
 How do I go about this?

**Comment by abentgeorge** [comment]
Apologies if I've used the wrong label, still new to using GitHub

**Comment by lildude** [comment]
This issue really should have been opened as a Discussion (last option in the list you were presented with when opening this issue). I'd transfer it, but the option isn't coming up and I'm still trying to workout why.

Please can you re-open this as a [discussion](https://github.com/github/linguist/discussions) in the "Language Analysis" category.

**Comment by lildude** [comment]
I've reopened this issue and will transfer it to discussions and answer once moved.

**Comment by lildude** [comment, marked as answer]
I've taken a look at your repository and I can see it is showing Jupyter notebook and Python:

![CleanShot 2021-01-22 at 17 33 19](https://user-images.githubusercontent.com/627280/105524718-ef73cd80-5cd7-11eb-917f-a164e90bb1e1.png)

If you click each of the languages in the side bar, you'll be able to see which files are responsible for each language.  The reason Juptyer is the primary language is because your single Jupyter file is significantly larger than your single python file (16.7kb vs 1.09kb) and Linguist works on the bytes of code.

If you would like Python to be the primary language, you'll need to either write more python 😉 or implement an override as detailed in in the [Overrides](https://github.com/github/linguist#overrides) section of the README.md

**Comment by abentgeorge** [comment]
Thanks alot Colin!
Gonna experiment with that! Decided to switch to Spyder as my IDE to avoid
this issue.

On Fri, Jan 22, 2021 at 11:07 PM Colin Seymour <notifications@github.com>
wrote:

> I've taken a look at your repository and I can see it is showing Jupyter
> notebook and Python:
>
> [image: CleanShot 2021-01-22 at 17 33 19]
> <https://user-images.githubusercontent.com/627280/105524718-ef73cd80-5cd7-11eb-917f-a164e90bb1e1.png>
>
> If you click each of the languages in the side bar, you'll be able to see
> which files are responsible for each language. The reason Juptyer is the
> primary language is because your single Jupyter file is significantly
> larger than your single python file (16.7kb vs 1.09kb) and Linguist works
> on the bytes of code.
>
> If you would like Python to be the primary language, you'll need to either
> write more python 😉 or implement an override as detailed in in the
> Overrides <https://github.com/github/linguist#overrides> section of the
> README.md
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/github/linguist/discussions/5162#discussioncomment-302975>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEABKGFM35SIAUVLR6GZPV3S3GZVVANCNFSM4WOZ5EUQ>
> .
>


-- 
With Regards,
Aben George
